### PR TITLE
feat: Use REST API handlers from sidetree-core-go

### LINKED ISSUE #78: Use REST API handlers from sidetree-core-go
Remove the REST API models from sidetree-mock and use the handlers/models in sidetree-core-go.

### PR DESCRIPTION
**Pull request by bstasyszyn**
Removed the REST API models and handlers from sidetree-mock and used the ones in sidetree-core-go.

closes #78

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>

### CODE REVIEW

File: test/bddtests/go.mod [hunk 1]
```diff
@@ -9,12 +9,9 @@ go 1.12
 require (
 	github.com/DATA-DOG/godog v0.7.13
 	github.com/fsouza/go-dockerclient v1.3.0
-	github.com/go-openapi/swag v0.19.0
 	github.com/pkg/errors v0.8.1
 	github.com/sirupsen/logrus v1.3.0
-	github.com/stretchr/testify v1.3.0
-	github.com/trustbloc/sidetree-core-go v0.0.0-20191017153620-a489e16494b3
-	github.com/trustbloc/sidetree-mock v0.0.0
+	github.com/trustbloc/sidetree-core-go v0.0.0-20191028174235-80e3b92d0da1
```
Comment (fqutishat) [inline test/bddtests/go.mod]: remove sidetree-mock from go.mod

Comment (troyronda) [reply test/bddtests/go.mod]: Where do you see sidetree-mock?

Comment (bstasyszyn) [reply test/bddtests/go.mod]: Done

File: test/bddtests/did_sidetree_steps.go [hunk 2]
```diff
@@ -15,22 +15,21 @@ import (
 	"time"
 
 	"github.com/trustbloc/sidetree-core-go/pkg/document"
+	"github.com/trustbloc/sidetree-core-go/pkg/restapi/model"
```
Comment (troyronda) [inline test/bddtests/did_sidetree_steps.go]: extra line

File: test/bddtests/go.mod [hunk 3]
```diff
@@ -9,13 +9,7 @@ go 1.12
...
 	github.com/DATA-DOG/godog v0.7.13
 	github.com/fsouza/go-dockerclient v1.3.0
-	github.com/go-openapi/swag v0.19.0
 	github.com/pkg/errors v0.8.1
 	github.com/sirupsen/logrus v1.3.0
-	github.com/stretchr/testify v1.3.0
-	github.com/trustbloc/sidetree-core-go v0.0.0-20191017153620-a489e16494b3
-	github.com/trustbloc/sidetree-mock v0.0.0
-
+	github.com/trustbloc/sidetree-core-go v0.0.0-20191028174235-80e3b92d0da1
```
Comment (troyronda) [inline test/bddtests/go.mod]: Should we be surprised that the bddtests don't refer to its parent project? @fqutishat @bstasyszyn

File: test/bddtests/did_sidetree_steps.go (hunk 2 above)
Comment (bstasyszyn) [reply test/bddtests/did_sidetree_steps.go]: Done

File: test/bddtests/go.mod (hunk 3 above)
Comment (bstasyszyn) [reply test/bddtests/go.mod]: I don't think so. The API is REST - no need to refer to the code. It still ends up testing the functionality.

**Pull request merged**